### ISSUE TITLE
[BUG] TabularToSeriesAdaptor inside larger transformer construction loses scikit-learn output configuration

### ISSUE THREAD

**Issue by kdekker-kdr4** [sktime/sktime#7333]
**Describe the bug**
Scikit-Learn based transformers allow to set the output format via the set_output api.
If such a transformer is wrapper in TabularToSeriesAdaptor the set output configuration remains. But if the wrapped transformer is used in a bigger pipeline this setting gets lost. (My expectation is due to the way sktime clones). The output of the scikit learn transformer becomes numpy again.

The TabularToSeriesAdaptor(input="pandas") does ensure that the wrapped transformer returns a dataframe, but not as should be with the set_output. This is via the sktime code and not via the scikit-learn set output mechanism. The TabularToSeriesAdaptor masks the set_output problem a bit, as only the output is wrapped in a dataframe and the index is set again. But the column names are forgotten.

**To Reproduce**

```python
from sklearn.impute import SimpleImputer
from sktime.transformations.compose import TransformerPipeline
from sktime.transformations.series.adapt import TabularToSeriesAdaptor
transformer_sklearn = SimpleImputer()
transformer_sklearn.set_output(transform="pandas")
transformer_sktime = TabularToSeriesAdaptor(transformer_sklearn, input_type="pandas")
transformer_pipeline = TransformerPipeline(steps=[transformer_sktime])
assert transformer.transformer_._sklearn_output_config["transform"] == "pandas"  # Use _sklearn_output_config["transform"] to retrieve the set output format.
assert transformer_pipeline.steps_[0][1].transformer_._sklearn_output_config["transform"] == "pandas"
```

**Expected behavior**
Both assert statements should return true. In other words the set_output configuration of a scikit-learn transformer should be copied over if TabularToSeriesAdaptor is used in a bigger transformer construction.

**Versions**
0.34.0

**Comment by fkiraly** [comment]
Curious, and interesting observation.

I believe though the problem is deeper - it is the *`sklearn`* `clone` that is problematic.

This will need some checking, but it seems that `sklearn` is inconsistently designed here, or at least it does not consider `set_output` as a clone-worthy field, because it is not handled like other parameters or configs.

In contrast, `scikit-base` (which `sklearn` does not use in `set_output`, instead it uses some custom code) does retain the values of all config fields in the `clone`, i.e., the values set by `set_config`.

The newest `sklearn` version adds a "hack" the clone method by explicitly overriding logic in the `clone` related to this one config field, at the end, since newest versions:
https://github.com/scikit-learn/scikit-learn/blob/d083972026dfaf14012d3d1752f0cced32645fa5/sklearn/base.py#L139

Since this is not programmatic but a one-case-step-out, this fragmentative change to the `sklearn` `clone` API has now diverged from `scikit-base`, which uses the programmatic entry point `set_config`, corresponding line:
https://github.com/sktime/skbase/blob/c8190b44a13922579e908ba4812211ed62883823/skbase/base/_base.py#L1730

I can see two solutions:

* add the line from `sklearn` in `skbase`. This feels very painful, as we would mix code from a bad design into a clean design.
* at the start of `skbase` `_clone`, add code that defines plug-ins. A basic design would be a list of check/action pairs that is followed, and the check could be "is this an sklearn estimator", and action would be "call sklearn `clone`". This would prevent further divergences, but still at the cost of fragmenting treatment of `sklearn` and non-`sklearn` estimators.
Thoughts, @kdekker-kdr4?

**Comment by fkiraly** [comment]
We can of course fix it in the `TabularToSeriesAdaptor` by setting it explicitly, that would solve the immediate/focused bug, but not the deeper issue.

Perhaps we could start with the quick fix ad a band-aid, and take some time to think about the deeper fix?

**Comment by kdekker-kdr4** [comment]
For me the fix in TabularToSeriesAdaptor would be sufficient as if I want to use an sklearn transformer I would use the TabularToSeriesAdaptor as wrapper. So fixing the clone there would be good enough for me. A wider solution might not be required as for sklearn estimators (with the make_reduction method  and the outputted class ) I do not face an issue regarding formatting.

If one of the longterm solutions need to be chose I would pick the second option. As my own experience is that you will need to taylor your code always a little bit to one of your main dependencies. A one fit clone method is very ambitious.

**Comment by fkiraly** [comment]
ok, I wanted to apply the quick fix, but I notice that we are already using `sklearn` `clone` inside the `TabularToSeriesAdaptor`.

So it is not entirely clear where the problem comes from - possibly cloning happens differently when recursive than when directly done as above. We need to think about the nesting logic carefully.

**Comment by fkiraly** [comment]
Here is another approach:
https://github.com/sktime/skbase/pull/380

We add a config that allows packages - or individual estimators like `TabularToSeriesAdaptor` - to set parameters that are retained.

**Comment by kdekker-kdr4** [comment]
I think that would solve this issue.

**Comment by fkiraly** [comment]
For good measure, I think we should also write a plugin system, so `sklearn` estimators use their own `clone`. Unfortunately, `clone` is not a method for `sklearn` estimators, but an external method. The dispatch architecture inside the `sklearn` `clone` is also quite hacky. A plugin system would solve all these issues.

**Comment by fkiraly** [comment]
Done here: https://github.com/sktime/skbase/pull/381, I think this is the better solution.

**Comment by fkiraly** [comment]
This should now be resolved, as `sktime 0.35.0` allows `scikit-base 0.12.0`, which contains the fix mentioned above. Please reopen if not, @kdekker-kdr4.
